binary_port/src/original_request_spec.rs

```diff
@@ -0,0 +1,83 @@
+use casper_types::bytesrepr::{self, Bytes, FromBytes, ToBytes};
+#[cfg(test)]
+use casper_types::testing::TestRng;
+#[cfg(test)]
+use rand::Rng;
+
+#[derive(Debug, PartialEq)]
+pub(crate) struct OriginalRequestSpec {
```

Comment:
I don't like the name `spec` here, because it's not really a specification for a request, it's more like context information
```suggestion
pub(crate) struct OriginalRequestContext
```

Reply:
Updated in caae87b0473987f31d965ca8e59cc3cac9b79ff2